client/src/main/java/io/pravega/client/stream/impl/ControllerImpl.java

```diff
@@ -254,38 +258,42 @@ public ControllerImpl(ManagedChannelBuilder<?> channelBuilder) {
 
     @Override
     public CompletableFuture<Boolean> scaleStream(final Stream stream, final List<Integer> sealedSegments,
+                                                  final Map<Double, Double> newKeyRanges, final long timeoutMillis,
```

Comment:
what i meant is even if check-scale was a longpoll, on the service we need to wait for the value to be updated so that we respond. And the value update can happen on a different controller instance than the one where the read request was received. 
So either we have a 'watch' like semantic on the store where this controller gets notified upon a value update in which case it should respond back to the caller. 
Otherwise, as i had said earlier, it will mean periodic store calls on the controller instance.